api/projects/tasks.py

```diff
@@ -22,3 +22,29 @@ def migrate_project_environments_to_v2(project_id: int) -> None:
                 IdentityOverridesV2MigrationStatus.COMPLETE
             )
             project.save()
+
+
+@register_task_handler()
+def handle_cascade_delete(project_id: int) -> None:
+    """
+    Due to the combination of soft deletes and audit log functionality,
+    cascade deletes for a project can take a long time for large projects.
+    This task delegates the cascade delete to tasks for the main related
+    entities for a project.
+    """
+
+    from environments.tasks import delete_environment
+    from features.tasks import delete_feature
+    from projects.models import Project
+    from segments.tasks import delete_segment
```

Comment:
I understand the local import for the `Project` import, but I'm surprised that the tasks were necessary to import this way.

Reply:
Ugh, yeah, I tried to resolve this but I ended up just diving into a hell hole of circular dependencies. It annoyingly made testing a little harder because I can't mock the imports, but I do feel like it's not a major issue for tasks to have local imports given their place in the stack.